### DISCUSSION TITLE
Best way to disable auto-format on save completely

### DISCUSSION THREAD

**Discussion by stoopkidddd** [LazyVim/LazyVim#141] [Q&A] [answered]
I see that the auto-command for format on save is being set in `plugins/lsp/format.lua` however I am not sure what the best way is to just stop that auto-command from being registered entirely. Any suggestions? I guess this question could apply to any default auto-command being set up

**Comment by folke** [comment, marked as answer]
You can't stop the autocmd from being created.
You can disable autoformat with the code below:
```lua
{
      "neovim/nvim-lspconfig",
       opts= {autoformat = false}
}
```

**Reply by jin-ahn** [reply]
How would I disable only for certain file types?

**Reply by jjohnston-aims** [reply]
How do you do this now with conform.nvim?

**Reply by folke** [reply]
- Auto-formatting can be disabled with:

    ```lua
    vim.g.autoformat = false -- globally
    vim.b.autoformat = false -- buffer-local
    ```

**Reply by odinho** [reply]
Doing it only for the session (so not answering the question, but this is top duckduckgo hit anyway):

```
<leader>uf
```

Easy to find when you figure out it's in 'ui', but still :)

**Reply by faulesocke** [reply]
How would I enable it only for specific languages? For example, I want autoformatting to be enabled in my Rust projects but literally nowhere else.

**Reply by dpetka2001** [reply]
Just do in your `/lua/config/options.lua` `vim.g.autoformat = false` and then in `/after/ftplugin/name_of_filetype.lua` put `vim.b.autoformat = true`. That should do what you want I believe. Or you could use an autocommand on `FileType` event and set the patterns you want there, so you don't have to create multiple files in `/after/ftplugin/` directory if you want to do it for more than one filetypes.

**Reply by surmish** [reply]
@faulesocke This is what I do in my `$HOME/.config/nvim/lua/config/autocmds.lua`

```lua
-- Autoformat setting
local set_autoformat = function(pattern, bool_val)
  vim.api.nvim_create_autocmd({ "FileType" }, {
    pattern = pattern,
    callback = function()
      vim.b.autoformat = bool_val
    end,
  })
end

set_autoformat({ "cpp" }, true)
set_autoformat({ "fish" }, false)
set_autoformat({ "lua" }, false)
set_autoformat({ "perl" }, false)
set_autoformat({ "yaml" }, false)
```

I don't know about the ordering of this w.r.t to `options.lua`.

**Reply by faulesocke** [reply]
Thanks for the quick answers! The `autocmds.lua` approach looks more promising because then I have it in one place for all languages.

**Reply by beholders-eye** [reply]
Thank you @surmish for the answer, helped me a lot too. 👏🏼

**Comment by HaleTom** [comment]
> nvim-lspconfig.opts.autoformat is deprecated. Please use vim.g.autoformat instead

LazyVim/lua/lazyvim/util/plugin.lua:

        Util.deprecate("nvim-lspconfig.opts.autoformat", "vim.g.autoformat")

LazyVim is by default setting it [here](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/options.lua#L6)

**Comment by HaleTom** [comment]
`lua/config/options.lua`:

```
-- Disable format on save
-- Undo LazyVim default: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/options.lua#L6
vim.g.autoformat = false
```

**Reply by lucas-marianno** [reply]
this is the correct way!!
